impl/src/main/java/org/jboss/weld/logging/EventLogger.java

```diff
@@ -102,4 +102,10 @@
     @LogMessage(level = Level.WARN)
     @Message(id = 419, value = "{0} is not a valid notification mode, using the default: {1}", format = Format.MESSAGE_FORMAT)
     void invalidNotificationMode(Object mode, Object defaultMode);
+
+    @Message(id = 420, value = "The implementation of ExecutorServices has to provide an instance of SchedulesExecutorServices!", format = Format.MESSAGE_FORMAT)
+    UnsupportedOperationException noScheduledExecutorServicesProvided();
+
+    @Message(id = 421, value = "Invalid input value for observer notification timeout. Has to be parseable String. Original exception: {0}", format = Format.MESSAGE_FORMAT)
```

Comment:
True, I was rather unsure if we should wrap the exception in `IllegalArgumentException` in the first place. The original NFE is pretty much self-explanatory in this case.